### PR TITLE
Integrate bundle-analyzer

### PR DESCRIPTION
**Pull request by koss-lebedev**
This PR integrates with https://www.bundle-analyzer.com/ to check that file sizes are within the specified limit. I put 200kb as a max size for JS bundle, and 100kb for CSS. We can adjust them later, but for now they look like a good starting point.

### CODE REVIEW

File: bundle-analyzer.config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  files: [
+    {
+      test: '*.js',
+      maxSize: '200 kB',
+    },
+    {
+      test: '*.css',
+      maxSize: '100 kB',
```
Comment (HonzaBartos) [inline bundle-analyzer.config.js:9]: @koss-lebedev Nothing but just out of curiosity - How did we come up with these `maxSize` values? Is it a standard? Thanks

Comment (koss-lebedev) [reply bundle-analyzer.config.js:9]: @HonzaBartos Well, the default setting for Bundle Analyzer is to cap all files at 250kb. I checked that the output from Gatsby for JS was under 200kb, so I thought it'd be nice to keep it that way 🙂The value for CSS size is random, just from experience, I think it should never grow over 100kb

File: README.md [hunk 2]
```diff
@@ -92,3 +92,12 @@ Looking for more guidance? Full documentation for Gatsby lives [on the website](
...
 
 [![Deploy to Netlify](https://www.netlify.com/img/deploy/button.svg)](https://app.netlify.com/start/deploy?repository=https://github.com/gatsbyjs/gatsby-starter-default)
+
+
+## Measurement tools
+
+### Bundle-analyzer
+
+The project integrates with [bundle-analyzer](https://www.bundle-analyzer.com/) to control the size of the build output, including JavaScript files, CSS stylesheets and images. BA will run a check on every PR and validate that the size of the build output is within the limits specified in [bundle-analyzer.config.js](/bundle-analyzer.config.js) file.
+
```
Comment (dannytce) [inline README.md:102]: I would add there a note, that the tool is for free.

The second note is that it shows some overview etc. on master branch, not on feature branch :)

File: bundle-analyzer.config.js (hunk 1 above)
Comment (dannytce) [reply bundle-analyzer.config.js:9]: Afaik a standard for JavaScript is 170kB :) 

I would leave there a note/todo, that this supposed to be ideally set in hand with performance budget. Even though this one kinda behaves as a perf budget, but it's not exactly it :)

File: README.md [hunk 3]
```diff
@@ -92,3 +92,12 @@ Looking for more guidance? Full documentation for Gatsby lives [on the website](
...
+To build project locally, you have to export `BUNDLE_ANALYZER_TOKEN` environment variable before running `yarn build`.
```
Comment (dannytce) [inline README.md:103]: Is there any way how to NOT run bundle-analyzer locally? Sometimes you would like to just try if it builds. Any extra operation takes extra time :) 

Perhaps let's add some flag, or have a special script: `"yarn build:analyze": "ANALYZE=true yarn build"`? :)

**Comment by dannytce** [comment]
@koss-lebedev could you also pretty please change the base of PR into v3? 🙏

**Comment by koss-lebedev** [comment]
@dannytce Good question 🙂To be honest, I didn't consider other options because bundle-analyzer was referenced directly in the issue, but if you don't mind, I would like to take a look at other options

**Comment by dannytce** [comment]
@koss-lebedev yeah originally I wasn't planning to do that. But even with pretty straightforward Lighthouse #114 I realized there are numerous implementations of the same thing. So I thought it might be worth to try them and document them :)

**Pull request merged**